app/scripts/modules/amazon/src/loadBalancer/configure/application/ConfigureRedirectConfigModal.tsx

```diff
@@ -0,0 +1,167 @@
+import * as React from 'react';
+import { Modal } from 'react-bootstrap';
+import { Formik, Form, FormikErrors } from 'formik';
+
+import {
+  TextInput,
+  FormikFormField,
+  ModalClose,
+  ReactModal,
+  SubmitButton,
+  noop,
+  SelectInput,
+  HelpField,
+  ReactSelectInput,
+} from '@spinnaker/core';
+
+import { IRedirectActionConfig } from 'amazon/domain';
+
+import './ConfigureConfigModal.css';
+
+export interface IConfigureRedirectConfigModalProps {
+  config: IRedirectActionConfig;
+  closeModal?(result?: any): void; // provided by ReactModal
+  dismissModal?(rejection?: any): void; // provided by ReactModal
+}
+
+export interface IConfigureRedirectConfigModalState {
+  initialValues: IRedirectActionConfig;
+}
+
+export class ConfigureRedirectConfigModal extends React.Component<
+  IConfigureRedirectConfigModalProps,
+  IConfigureRedirectConfigModalState
+> {
+  public static defaultProps: Partial<IConfigureRedirectConfigModalProps> = {
+    closeModal: noop,
+    dismissModal: noop,
+  };
+
+  public static show(props: IConfigureRedirectConfigModalProps): Promise<void> {
+    return ReactModal.show(ConfigureRedirectConfigModal, props);
+  }
+
+  constructor(props: IConfigureRedirectConfigModalProps) {
+    super(props);
+
+    const config = props.config || ({} as IRedirectActionConfig);
+
+    this.state = {
+      initialValues: {
+        host: config.host || '',
+        path: config.path || '',
+        port: config.port || '',
+        protocol: config.protocol || undefined,
+        query: config.query || '',
+        statusCode: config.statusCode || 'HTTP_301',
+      },
+    };
+  }
+
+  private close = (): void => {
+    this.props.dismissModal.apply(null, arguments);
+  };
+
+  private submit = (data: IRedirectActionConfig): void => {
+    const filteredData = Object.keys(data).reduce(
+      (acc, key: keyof IRedirectActionConfig) => {
+        if (data[key] && data[key] !== '') {
+          acc[key] = data[key];
+        }
+        return acc;
+      },
+      {} as IRedirectActionConfig,
+    );
+    this.props.closeModal(filteredData);
+  };
+
+  private validate = (): FormikErrors<IRedirectActionConfig> => {
+    return {};
+  };
+
+  public render() {
+    const { initialValues } = this.state;
```

Comment:
FWIW, if an object doesn’t participate in the render lifecycle such as `initialValues`, it’s OK to put it as a member property.  Putting it in `state` implies that it will change

Reply:
That is a true statement.